### PR TITLE
feat: Sendable collections improvements

### PR DESCRIPTION
**Pull request by PhilippeWeidmann**
I chose to not reimplement the whole `RangeReplaceableCollection` protocol but rather add methods when needed.

### CODE REVIEW

File: Tests/InfomaniakCoreTests/UTCollectionTests.swift [hunk 1]
```diff
@@ -94,6 +94,41 @@ final class UTSendableArray: XCTestCase {
...
+
+        // WHEN
+        let t = Task.detached {
+            collection.removeAll()
+        }
+
+        _ = await t.result
+
+        // THEN
+        XCTAssertEqual(collection.count, 0)
```
Comment (adrien-coye) [inline Tests/InfomaniakCoreTests/UTCollectionTests.swift:112]:
```suggestion
        XCTAssertTrue(collection.isEmpty)
```

Comment (adrien-coye) [reply Tests/InfomaniakCoreTests/UTCollectionTests.swift:112]: Using more specific `XCTAssertX` helps understanding the test the day it fails and you have to understand what broke.

File: Tests/InfomaniakCoreTests/UTCollectionTests.swift [hunk 2]
```diff
@@ -94,6 +94,41 @@ final class UTSendableArray: XCTestCase {
...
+
+        // WHEN
+        let t = Task.detached {
+            collection.removeAll(where: { $0 == "b" })
+        }
+
+        _ = await t.result
+
+        // THEN
+        XCTAssertEqual(collection.values.contains("b"), false)
```
Comment (adrien-coye) [inline Tests/InfomaniakCoreTests/UTCollectionTests.swift:130]:
```suggestion
        XCTAssertFalse(collection.values.contains("b"))
```

File: Tests/InfomaniakCoreTests/UTCollectionTests.swift [hunk 3]
```diff
@@ -129,4 +164,21 @@ final class UTSendableDictionary: XCTestCase {
...
+
+        // WHEN
+        let t = Task.detached {
+            collection.removeAll()
+        }
+
+        _ = await t.result
+
+        // THEN
+        XCTAssertEqual(collection.count, 0)
```
Comment (adrien-coye) [inline Tests/InfomaniakCoreTests/UTCollectionTests.swift:182]:
```suggestion
        XCTAssertTrue(collection.isEmpty)
```

**Review by adrien-coye — COMMENTED** [review]
I just have a small proposal regarding how we check assertions.

**Comment by adrien-coye** [comment]
Note: Something I do not really like today is the fact SendableCollections are classes. I mean it does work but you can write a `let` and mutate it, breaking with expected swift collections behaviour. Just saying that might be something else we could improve.

**Comment by PhilippeWeidmann** [comment]
> Note: Something I do not really like today is the fact SendableCollections are classes (For the record I wrote it this way for convenience). I mean it does work but you can write a `let` and mutate it, breaking with expected swift collections behaviour. Just saying that might be something else we could improve.

Yup totally agree

**Review by adrien-coye — APPROVED** [review]
Thanks for updating the code.

**Pull request merged**